File: Java/dependency-injection/_05_service-locator/src/main/java/ServiceLocatorAntipattern/Core/TeleComSystem.java
package ServiceLocatorAntipattern.Core;

import ServiceLocatorAntipattern.ApplicationRoot;
import ServiceLocatorAntipattern.Inbound.Inbound;
import ServiceLocatorAntipattern.Outbound.Outbound;

public class TeleComSystem {
  private final ProcessingWorkflow processingWorkflow;
  private final Inbound inbound;
  private final Outbound outbound;

  public TeleComSystem() {
    inbound = ApplicationRoot.context.getComponent(Inbound.class);
    outbound = ApplicationRoot.context.getComponent(Outbound.class);
    processingWorkflow = ApplicationRoot.context.getComponent(ProcessingWorkflow.class);
  }

  public void start() {
    inbound.setDomainLogic(processingWorkflow);
    processingWorkflow.setOutbound(outbound);
    inbound.startListening();
  }
}

File: CSharp/DependencyInjection/DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Core/TeleComSystem.cs
using ServiceLocatorDIAntipattern._2_ServiceLocator.Inbound;
using ServiceLocatorDIAntipattern._2_ServiceLocator.Outbound;
using Unity;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Core
{
  class TeleComSystem
  {
    private readonly IProcessingWorkflow _processingWorkflow;
    private readonly IInbound _inbound;
    private readonly IOutbound _outbound;

    public TeleComSystem()
    {
      _inbound = ApplicationRoot.Context.Resolve<IInbound>();
      _outbound = ApplicationRoot.Context.Resolve<IOutbound>();
      _processingWorkflow = ApplicationRoot.Context.Resolve<IProcessingWorkflow>();
    }

    public void Start()
    {
      _inbound.SetDomainLogic(_processingWorkflow);
      _processingWorkflow.SetOutbound(_outbound);
      _inbound.StartListening();
    }
  }
}